nuxeo-core/nuxeo-core-storage/src/main/java/org/nuxeo/ecm/core/storage/kv/MemKeyValueStore.java

```diff
@@ -33,24 +33,26 @@
  */
 public class MemKeyValueStore implements KeyValueStoreProvider {
 
-    protected final Map<String, byte[]> map;
-
-    protected final Lock readLock;
+    protected final ExpiringMap<String, byte[]> map;
 
     protected final Lock writeLock;
 
+    protected String name;
+
     public MemKeyValueStore() {
-        ReadWriteLock rwLock = new ReentrantReadWriteLock();
-        readLock = rwLock.readLock();
-        writeLock = rwLock.writeLock();
-        // We can't use ConcurrentHashMap because compareAndSet needs to compare by value
-        // and ConcurrentHashMap doesn't know how to do that. Instead we use full locking;
-        // this is ok as this class isn't expected to be used in a high write rate scenario.
-        map = new HashMap<>();
+        map = ExpiringMap.builder().expiration(Integer.MAX_VALUE, TimeUnit.DAYS).variableExpiration().build();
```

Comment:
if there is no capacity limits or no eviction with a default ttl,  we may warn in the javadoc that this can impact the application performance and for large volume other implementation are prefered

Reply:
No there's no JMX metrics exposed.